interface/src/app/gene/network/network.js

```diff
@@ -39,71 +40,92 @@ angular.module('adage.gene.network', [
   });
 }])
 
-.factory('EdgeService', ['$resource', function($resource) {
+.factory('EdgeService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
   // Possible parameters for this endpoint when making a query are:
   // {
   //   genes: Database IDs of genes associated with the edge,
   //   mlmodel: Database ID of MLModel associated with the edge,
   //   limit: Maximum number of results to return
   //  }
-  return $resource('/api/v0/edge/');
-}])
+    return $resource(ApiBasePath + 'edge');
+  }
+])
 
-.factory('NodeService', ['$resource', function($resource) {
-  return $resource('/api/v0/node/');
-}])
+.factory('NodeService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
+    return $resource(ApiBasePath + 'node');
+  }
+])
 
 .controller('GeneNetworkCtrl',
-   ['$stateParams', 'EdgeService', 'NodeService', '$log',
+  ['$stateParams', 'EdgeService', 'NodeService', '$log', 'errGen',
     function GeneNetworkController($stateParams, EdgeService, NodeService,
-                                   $log) {
-      var rawMinWeight = -1.0;
-      var rawMaxWeight = 1.0;
-      var minCorrelation = rawMinWeight;
-      var scaledMaxWeight = rawMaxWeight - rawMinWeight;
+                                   $log, errGen) {
+      var self = this;
+      // Do nothing if no genes are specified in URL.
+      if (!$stateParams.genes || !$stateParams.genes.split(',').length) {
+        self.statusMessage = 'No genes are specified.';
+        return;
+      }
+
+      // The following properties of "self" will be available to HTML.
+      self.includePositive = true;
+      self.includeNegative = true;
+      self.statusMessage = 'Connecting to the server ...';
+
+      var minCorrelation = -1.0, maxCorrelation = 1.0;
+      var midPoint = (minCorrelation + maxCorrelation) / 2.0;
       var setEdgeColor = d3.scale.linear()
-          .domain([0, scaledMaxWeight / 2.0, scaledMaxWeight])
+          .domain([minCorrelation, midPoint, maxCorrelation])
           .range(['green', 'orange', 'red']);
 
       var network = d3.network()  // Initialize the network.
-          .minEdge(0)
-          .maxEdge(rawMaxWeight - rawMinWeight)
+          .minEdge(minCorrelation)
+          .maxEdge(maxCorrelation)
           .edgeColor(setEdgeColor)
           .geneText(function(d) {
             return d.label;
           })
-          .legendStart(rawMinWeight)
-          .legendEnd(rawMaxWeight)
+          .legendStart(minCorrelation)
+          .legendEnd(maxCorrelation)
           .legendText('Correlation');
 
-      var self = this;
-      // The following properties of "self" will be available to HTML.
-      self.maxNodeNum = Number.MAX_SAFE_INTEGER;
-      self.statusMessage = 'Connecting to the server ...';
-      self.slider = {  // Slider configuration
-        position: minCorrelation,
+      var geneTip, edgeTip;
+
+      self.updateSvg = function() {
```

Comment:
Minor quibble: can we name this function something closer to its purpose within the scope of `network.js`? I know the network diagram is drawn using SVG, but that is an implementation detail and none of the actual code inside this function actually does anything to manipulate SVG. I'm thinking the right abstraction level would be something like `renderNetwork` or `redrawNetwork`.